counsel.el

```diff
@@ -2027,14 +2041,15 @@ The preselect behavior can be customized via user options
               :caller caller)))
 
 ;;;###autoload
-(defun counsel-find-file (&optional initial-input)
+(defun counsel-find-file (&optional initial-input initial-directory)
   "Forward to `find-file'.
 When INITIAL-INPUT is non-nil, use it in the minibuffer during completion."
   (interactive)
   (counsel--find-file-1
    "Find file: " initial-input
    #'counsel-find-file-action
-   'counsel-find-file))
+   'counsel-find-file
+   initial-directory))
```

Comment:
Instead of passing this as a new argument to `counsel--find-file-1`, why not bind `default-directory` around the call to it?

Reply:
That sounds like a fine idea to me, though I don't know how to go about doing that. Any direction on this?

Reply:
Here's what I mean:

```diff
diff --git a/counsel.el b/counsel.el
index 0e6a26f2..34292c58 100644
--- a/counsel.el
+++ b/counsel.el
@@ -2031,10 +2031,10 @@ counsel-find-file
   "Forward to `find-file'.
 When INITIAL-INPUT is non-nil, use it in the minibuffer during completion."
   (interactive)
-  (counsel--find-file-1
-   "Find file: " initial-input
-   #'counsel-find-file-action
-   'counsel-find-file))
+  (let ((default-directory initial-directory))
+    (counsel--find-file-1 "Find file: " initial-input
+                          #'counsel-find-file-action
+                          'counsel-find-file)))
 
 (ivy-configure 'counsel-find-file
   :parent 'read-file-name-internal
```

Reply:
I see better what you mean, looks good to me.